lms/services/mailchimp.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from dataclasses import dataclass
+
+import mailchimp_transactional
+
+LOG = logging.getLogger(__name__)
+
+
+@dataclass
+class EmailSender:
+    subaccount: str
+    """The Mailchimp subaccount ID to use to send the email."""
+
+    email: str
+    """The email address that the email will appear to come from."""
+
+    name: str
+    """The full name that the email will appear to come from."""
+
+
+@dataclass
+class EmailRecipient:
+    email: str
+    """The email address to send to."""
+
+    name: str
+    """The recipient full name to use in the email's To: header."""
+
+
+class MailchimpService:
+    def __init__(self, api_key):
+        if api_key:
+            self.mailchimp_client = mailchimp_transactional.Client(api_key)
+
+    def send_template(
+        self,
+        template_name: str,
+        sender: EmailSender,
+        recipient: EmailRecipient,
+        template_vars: dict,
+    ):
+        """
+        Send an email using Mailchimp Transactional's send-template API.
+
+        https://mailchimp.com/developer/transactional/api/messages/send-using-message-template/
+        """
+        params = {
+            "template_name": template_name,
+            "message": {
+                "subaccount": sender.subaccount,
+                "from_email": sender.email,
+                "from_name": sender.name,
+                "to": [{"email": recipient.email, "name": recipient.name}],
+                "track_opens": True,
+                "track_clicks": True,
+                "global_merge_vars": [
+                    {"name": key, "content": value}
+                    for key, value in template_vars.items()
+                ],
```

Comment:
The Mailchimp client requires the template variables ("merge vars") to be in a weird format. Instead of just a dict it's a list of dicts in which each dict has two keys `"name"` (the variable name) and `"content"` (the variable value). I've made it so that we can just pass a dict to `MailchimpService.send_template()` and it'll translate it into Mailchimp's awkward format